LIS3MDL.device.nut

```diff
@@ -0,0 +1,236 @@
+// Copyright (c) 2015 Electric Imp
+// This file is licensed under the MIT License
+// http://opensource.org/licenses/MIT
+
+class LIS3MDL {
+
+    static VERSION = [1,0,0];
+
+    // External constants
+    static AXIS_X = 0x80;
+    static AXIS_Y = 0x40;
+    static AXIS_Z = 0x20;
+    // Note that if INTERRUPT_ISACTIVEHIGH is used to generate a wake signal, significant power may be wasted.  See README for details.
+    static INTERRUPT_ISACTIVEHIGH = 0x04;
+    static INTERRUPT_DONTLATCH = 0x02;
+    static DATA_RATE_FAST = -1;
+    static CONVERSION_TYPE_CONTINUOUS = 0x00;
+    static CONVERSION_TYPE_SINGLE = 0x01;
+
+    // Internal constants
+    static REG_ADDR_OUT_X_L = 0x28;
+    static REG_ADDR_OUT_Y_L = 0x2A;
+    static REG_ADDR_OUT_Z_L = 0x2C;
+    static REG_CTL_1 = 0x20;
+    static REG_CTL_2 = 0x21;
+    static REG_CTL_3 = 0x22;
+    static REG_CTL_4 = 0x23;
+    static REG_STATUS = 0x27;
+    static REG_INT_CFG = 0x30;
+    static REG_INT_SRC = 0x31;
+    static REG_INT_THS_L = 0x32;
+    static REG_INT_THS_H = 0x33;
+    static SENSITIVITY_OF_MIN_SCALE = 27368.0; // = (4 gauss scale) * (6842 LSB/gauss at 4 gauss scale)
+
+    _i2c = null;
+    _address = null;
+    _scale = null;
+
+    function constructor(i2c, address=0x1C) {
+        _i2c = i2c;
+        _address = address;
+
+        init();
+    }
+
+    function init() {
```

Comment:
It's necessary for when the device is power-cycled/reset out of band with the class.  This method updates the internal reading scale from the device.